### PR TITLE
[proxima-beam-core] #197 use unnamed consumer in not named explicitly

### PR DESCRIPTION
**Pull request by je-ik**

### CODE REVIEW

File: beam/core/src/main/java/cz/o2/proxima/beam/direct/io/CommitLogRead.java [hunk 1]
```diff
@@ -531,8 +527,10 @@ public Instant getInitialWatermarkEstimatorState() {
   }
 
   @VisibleForTesting
-  BlockingQueueLogObserver newObserver(String name, OffsetRange restriction) {
+  BlockingQueueLogObserver newObserver(@Nullable String name, OffsetRange restriction) {
     return BlockingQueueLogObserver.create(
-        name, restriction.getTotalLimit(), Watermarks.MIN_WATERMARK);
+        name != null ? name : UUID.randomUUID().toString(),
```
Comment (LesTR) [inline beam/core/src/main/java/cz/o2/proxima/beam/direct/io/CommitLogRead.java:532]: Why can't we leave handling this for concrete `CommitLogReader` implementation as it is now? For example [here](https://github.com/datadrivencz/proxima-platform/blob/master/direct/io-pubsub/src/main/java/cz/o2/proxima/direct/pubsub/PubSubReader.java#L260) we already have `@Nullable` in name contract.

Comment (LesTR) [reply beam/core/src/main/java/cz/o2/proxima/beam/direct/io/CommitLogRead.java:532]: Och, I see it now, It's related to `BlockingQueueLogObserver`

**Review by LesTR — APPROVED** [review]
LGTM 👍

**Comment by je-ik** [comment]
This closes O2-Czech-Republic#197

**Pull request merged**